### ISSUE TITLE
How to turn on localhost data server

### ISSUE THREAD

**Issue by xzhub** [nextstrain/nextclade#853]
I changed the `DATA_FULL_DOMAIN` from 'https://data.master.clades.nextstrain.org' to 'http://localhost:27722' in my .env, since I'd like to boot up a nextclade server in our local environment. 

However, this change makes the server inaccessible. I only see an icon spinning, nothing else. 

How can I turn on the local data server? I think there is no service on local port 27722.

my commnds to build the server:

```
export VER=2.0.0-alpha.5

# build 
cd /nextclade-${VER}/packages_rs/nextclade-web && yarn install && \
    yarn wasm-prod && \
    yarn prod:build

# clean
yarn autoclean --init && yarn autoclean --force && yarn cache clean

# download data
mkdir /nextclade-${VER}/packages_rs/nextclade-web/data && \
    chmod 777 /nextclade-${VER}/packages_rs/nextclade-web/data && \
    nextclade dataset get --server="https://data.master.clades.nextstrain.org" \
        --name='sars-cov-2' --output-dir=/nextclade-${VER}/packages_rs/nextclade-web/data/sars-cov-2

# start server
yarn prod:serve
```

**Comment by ivan-aksamentov** [comment]
@xzhub You cannot just download one dataset and expect Nextclade to find it. In order to discover available datasets (for `dataset list` CLI command or for the dataset selector in web app)  Nextclade expects an index JSON file. So it is more complicated.

Production dataset server is just a static web server: an S3 bucket + Cloudfront cache. It serves the result produced by a GitHub Action in the data repo: https://github.com/nextstrain/nextclade_data

In order to reproduce the same thing locally, you can:
 1. Build dataset index file from the data repo using `./scripts/rebuild`
 2. Serve `data_output/` directory using any static file server
 3. Set the address of this server to `DATA_FULL_DOMAIN` or use `--server` flag in CLI, so that Nextclade can find it

One example is given in the developer guide in the section "Trying custom datasets locally":
https://github.com/nextstrain/nextclade/blob/master/docs/dev/developer-guide.md#trying-custom-datasets-locally

This is just one example. Theoretically, Nextclade only cares about index file being in the right format and that all the file URLs in it are accessible. You can serve anything you want from anywhere.

**Comment by xzhub** [comment]
For some reason I didn't make it work in the 2.0.0 alpha version with local server, the icon keep spinning and I can't see the home page.
With the latest released 2.2.0 version, I successfully switched to the local data server. One thing I'd like to add is that the local server need to turn on CORS, for example in nginx's config:

```
    location / {
        add_header Access-Control-Allow-Origin *;  # turn on CORS
        root   /usr/share/nginx/html;
        index  index.html index.htm;
    }
```

**Comment by ivan-aksamentov** [comment]
@xzhub Yes, CORS is needed. It is [mentioned](https://github.com/nextstrain/nextclade/blob/master/docs/dev/developer-guide.md#trying-custom-datasets-locally) in the example with `serve` NPM package in the docs as a `--cors` flag:
```bash
npx serve --cors --listen=tcp://0.0.0.0:27722 data_output/
```
but it would be nice to add a note about CORS in plain English as well.

To prevent infinite logo spinner on errors, I added some changes after official release 2.0.0, so now it should hopefully show error popups if dataset index fetch fails. Plus there's detailed messages in the browser console.
